### DISCUSSION TITLE
Adjust mask size and padding pixels based on percentage.

### DISCUSSION THREAD

**Discussion by nykwil** [kex0/batch-face-swap#21] [Ideas]
I find I want to do the whole head. But if I set the mask size to 10 it can very wildly how much that is depending on the resolution. What I really want is to adjust the mask based on the face size percentage. 

Like If the face takes up 100x100 then grow it by 10% to 110 but 20x20 grow to 22.

I also wish it could control the padding pixels automatically in the same way so the face is always the same size for i2i.

**Comment by nothings** [comment]
Not the main dev here, but I have plans to eventually try to write an explicit "head mode" that just grows the mask by some fraction of the face size, but tries to prevent the masks from growing into each other. But maybe there's something simpler in the interim that can be done.
